autoload/lsp/lspserver.vim

```diff
@@ -1159,7 +1159,18 @@ def TextDocFormat(lspserver: dict<any>, fname: string, rangeFormat: bool,
 				start_lnum: number, end_lnum: number)
   # Check whether LSP server supports formatting documents
   if !lspserver.isDocumentFormattingProvider
-    util.ErrMsg('LSP server does not support formatting documents')
+    if &formatexpr !=# 'lsp#lsp#FormatExpr()' && opt.lspOptions.formatFallback
```

Comment:
Instead of duplicating this code in lsp.vim and lspserver.vim, can you move this to a function in util.vim and use that?